test/integration/client-side-encryption/range_explicit_encryption.prose.test.ts

```diff
@@ -0,0 +1,542 @@
+        }
+      });
+
+      it('Case 4: can find encrypted range with an open range query', metaData, async function () {
+        const query = {
+          $and: [{ 'encrypted<Type>': { $gt: factory(30) } }]
+        };
+
+        const findPayload = await clientEncryption.encryptExpression(query, {
+          keyId,
+          algorithm: 'RangePreview',
+          queryType: 'rangePreview',
+          contentionFactor: 0,
+          rangeOptions
+        });
+
+        const key = `encrypted${dataType}`;
+        const result = (
+          await encryptedClient
+            .db('db')
+            .collection('explicit_encryption')
+            .find(findPayload)
+            .toArray()
+        ).map(doc => ({ _id: doc._id, [key]: doc[key] }));
+
+        result.sort(byId);
+
+        expect(result).to.have.lengthOf(1);
+
+        expect(result[0]).to.have.property('_id', 3);
+        compareNumericValues(result[0][key], 200);
+      });
+
+      it('Case 5: can run an aggregation expression inside $expr', metaData, async function () {
+        const query = { $and: [{ $lt: ['$encrypted<Type>', factory(30)] }] };
+
+        const findPayload = await clientEncryption.encryptExpression(query, {
+          keyId,
+          algorithm: 'RangePreview',
+          queryType: 'rangePreview',
+          contentionFactor: 0,
+          rangeOptions
+        });
+
+        const key = `encrypted${dataType}`;
+        const result = (
+          await encryptedClient
+            .db('db')
+            .collection('explicit_encryption')
+            .find({ $expr: findPayload })
+            .toArray()
+        ).map(doc => ({ _id: doc._id, [key]: doc[key] }));
+
+        result.sort(byId);
+
+        const expected = [
+          {
+            [key]: 0,
+            _id: 0
+          },
+          {
+            [key]: 6,
+            _id: 1
+          }
+        ];
+
+        expect(result).to.have.lengthOf(expected.length);
+
+        for (let i = 0; i < expected.length; ++i) {
+          const doc = result[i];
+          const expectedDoc = expected[i];
+          expect(doc).to.have.property('_id', expectedDoc['_id']);
+          compareNumericValues(doc[key], expectedDoc[key]);
+        }
+      });
+
+      it(
+        'Case 6: encrypting a document greater than the maximum errors',
+        metaData,
+        async function () {
+          if (dataType === 'DoubleNoPrecision' || dataType === 'DecimalNoPrecision') {
+            this.test.skipReason =
+              'Case 6 does not apply to DoubleNoPrecision or DecimalNoPrecision';
+            this.skip();
+          }
+          const resultOrError = await clientEncryption
+            .encrypt(factory(201), {
+              keyId,
+              algorithm: 'RangePreview',
+              contentionFactor: 0,
+              rangeOptions
+            })
+            .catch(e => e);
+
+          expect(resultOrError).to.be.instanceOf(Error);
```

Comment:
We either want to check MongoCryptError here, or a portion of the error message, this is broad enough to accept any kind of error and could be overly permissive